src/app/modules/chat/messages/messages.component.ts

```diff
@@ -35,14 +37,25 @@ export class MessagesComponent implements OnInit, AfterViewChecked {
 
     this.store.dispatch(new LoadMessagesBegin());
     this.store.dispatch(new GetCurrentUser());
+    this.store.dispatch(new StartOnlineOfflineCheck());
+
+    this.isOnline$ = this.store.select(selectOnlineOfflineCheck);
```

Comment:
Component should know nothing about current internet connection status. Move this logic out of here.